packages/near-fast-auth-signer/src/utils/multi-chain/signature/signature.ts

```diff
@@ -0,0 +1,152 @@
+import BN from 'bn.js';
+import { ethers } from 'ethers';
+import { Account, transactions } from 'near-api-js';
+
+import { RSVSignature } from './types';
+import { parseSignedDelegateForRelayer } from '../relayer';
+
+export type ChainSignatureContracts = 'multichain-testnet-2.testnet'
+
+const toRVS = (signature: string): RSVSignature => {
+  const parsedJSON = JSON.parse(signature) as [string, string];
+
+  return {
+    r: parsedJSON[0].slice(2),
+    s: parsedJSON[1],
+  };
+};
+
+/**
+ * Signs a transaction hash using a specified account and path, then sends the signed transaction
+ * to a relayer service for execution. It attempts to fetch the signature from the transaction
+ * receipt up to 3 times with a delay of 10 seconds between each attempt.
+ *
+ * @param {string | ethers.BytesLike} transactionHash - The hash of the transaction to be signed.
+ * @param {string} path - The derivation path used for signing the transaction.
+ * @param {Account} account - The NEAR account object used for signing the transaction.
+ * @param {string} relayerUrl - The URL of the relayer service to which the signed transaction is sent.
+ * @param {ChainSignatureContracts} contract - The contract identifier for chain signature operations.
+ * @returns {Promise<RSVSignature>} A promise that resolves to the RSV signature of the signed transaction.
+ * @throws {Error} Throws an error if the signature cannot be fetched after 3 attempts.
+ */
+export const sign = async (
+  transactionHash: string | ethers.BytesLike,
+  path: string,
+  account: Account,
+  relayerUrl: string,
+  contract: ChainSignatureContracts
+): Promise<RSVSignature> => {
+  const functionCall = transactions.functionCall(
+    'sign',
+    {
+      payload: Array.from(ethers.getBytes(transactionHash)).slice().reverse(),
+      path,
+    },
+    new BN('300000000000000'),
+    new BN(0)
```

Comment:
These are repeated, lets pull them out into constants above with appropriate name to give a bit more context what they are?
`NEAR_MAX_GAS` or something similar

Reply:
Agreed, will update it :)